src/widgets/MultiSelectList.js

```diff
@@ -0,0 +1,245 @@
+    let data = options
+      .filtered(queryString, queryText)
+      .sorted(sortByString)
+      .slice();
+
+    if (queryStringSecondary) {
+      data = data.concat(
+        complement(options.filtered(queryStringSecondary, queryText).sorted(sortByString), data)
+      );
+    }
+
+    return data;
+  };
+
+  /**
+   * Filters an array by two filter properties, and user input query text.
+   * Ignores case. Querying a realm result with filtered is more performant,
+   * so have two cases for each.
+   */
+  const filterArrayData = () => {
+    const regexFilter = RegExp(queryText, 'i');
+
+    return options.filter(
+      optionItem =>
+        regexFilter.test(optionItem[primaryFilterProperty]) ||
+        regexFilter.test(optionItem[secondaryFilterProperty])
+    );
+  };
+
+  /**
+   * Delegator of filtering process. Check if the object is a realm
+   * object (has the filtered member) or if it is an array. Otherwise,
+   * return an empty list to display.
+   */
+  const getData = () => {
+    if (options && options.filtered && queryString) return filterResultData(options);
+    if (Array.isArray(options) && primaryFilterProperty) return filterArrayData(options);
+    return [];
+  };
+
+  /**
+   * Render single item of the FlatList
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {component} ResultRow
+   *
+   */
+  const renderItem = useCallback(data => (
+    <ResultRow
+      data={data}
+      onPress={onSelect}
+      isSelected={isSelected}
+      renderLeftText={renderLeftText}
+      renderRightText={renderRightText}
+      showCheckIcon={showCheckIcon}
+    />
+  ));
+
+  /**
+   * On select FlatList item update selected state
+   *
+   * @prop  {object}     data   Single item data.
+   */
+  const onSelect = useCallback(data =>
+    setSelected(prevState =>
+      prevState.indexOf(data.item.id) === -1
+        ? [...prevState, data.item.id]
+        : prevState.filter(i => i !== data.item.id)
+    )
+  );
+
+  const EmptyComponent = useCallback(({ title }) => (
+    <View style={localStyles.emptyContainer}>
+      <Text style={localStyles.emptyText}>{title}</Text>
+    </View>
+  ));
+
+  /**
+   * Check if the single item is selected or not
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {bool}    selected state of the item
+   */
+  const isSelected = useCallback(data => !(selected.indexOf(data.item.id) === -1));
+
+  const onDoneSelected = useCallback(() => onConfirmSelections(selected));
+
+  const data = getData();
+
+  return (
+    <View style={localStyles.container}>
+      <SearchBar
+        autoCapitalize="none"
+        autoCorrect={false}
+        autoFocus
+        color="white"
+        onChange={setQueryText}
+        placeholder={placeholderText}
+        placeholderTextColor="white"
+        style={[localStyles.text, localStyles.searchBar]}
+      />
+      <FlatList
+        data={data}
+        keyExtractor={item => item.id || item.name}
```

Comment:
still various annonymous functions and new references to styles scattered thoughout this component and `ResultRow` - Trying to make the default of all code to be that these things are defined explicitly in a way where references aren't re-created as these can cause basic performance problems with re-rendering. It's fine for this component as it won't have performance issues